### PR TITLE
feat(crons): Add thresholds to monitor config payload

### PR DESCRIPTION
**Pull request by rjo100**
Adds `failure_issue_threshold` and `recovery_threshold` into `MonitorConfig` payload. Allows for SDKs to set thresholds programmatically via upsert.

### CODE REVIEW

File: relay-monitors/src/lib.rs [hunk 1]
```diff
@@ -97,14 +97,22 @@ pub struct MonitorConfig {
...
     #[serde(default, skip_serializing_if = "Option::is_none")]
     timezone: Option<String>,
+
+    /// How many consecutive failed check-ins it takes to create an issue
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    failure_issue_threshold: Option<u64>,
+
+    /// How many consecutive OK check-ins it takes to resolve an issue
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    recovery_threshold: Option<u64>,
```
Comment (evanpurkhiser) [inline relay-monitors/src/lib.rs]: Let's make sure to add these to the develop.sentry.dev SDK docs as well.

**Review by evanpurkhiser — APPROVED** [review]

File: relay-monitors/src/lib.rs [hunk 2]
```diff
@@ -97,14 +97,22 @@ pub struct MonitorConfig {
...
+    /// How long (in minutes) is the checkin allowed to run for in in_progress before it is
     /// considered failed.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     max_runtime: Option<u64>,
 
     /// tz database style timezone string
     #[serde(default, skip_serializing_if = "Option::is_none")]
     timezone: Option<String>,
+
+    /// How many consecutive failed check-ins it takes to create an issue
```
Comment (jjbayer) [inline relay-monitors/src/lib.rs]:
```suggestion
    /// How many consecutive failed check-ins it takes to create an issue.
```

File: relay-monitors/src/lib.rs [hunk 3]
```diff
@@ -97,14 +97,22 @@ pub struct MonitorConfig {
...
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    failure_issue_threshold: Option<u64>,
+
+    /// How many consecutive OK check-ins it takes to resolve an issue
```
Comment (jjbayer) [inline relay-monitors/src/lib.rs]:
```suggestion
    /// How many consecutive OK check-ins it takes to resolve an issue.
```

**Review by jjbayer — APPROVED** [review]
LGTM

File: relay-monitors/src/lib.rs [hunk 4]
```diff
@@ -97,14 +97,22 @@ pub struct MonitorConfig {
     #[serde(default, skip_serializing_if = "Option::is_none")]
     checkin_margin: Option<u64>,
 
-    /// How long (in minutes) is the checkin allowed to run for in in_rogress before it is
+    /// How long (in minutes) is the checkin allowed to run for in in_progress before it is
```
Comment (olksdr) [inline relay-monitors/src/lib.rs]:
```suggestion
    /// How long (in minutes) is the check-in allowed to run for in in_progress before it is
```

**Review by olksdr — COMMENTED** [review]
It would be great to keep the naming consistent throughout the docs, `checkin` vs. `check-in`

**Pull request merged**